lightning/src/ln/channelmanager.rs

```diff
@@ -10929,7 +10929,18 @@ where
 		let events = core::cell::RefCell::new(Vec::new());
 		let event_handler = |event: events::Event| Ok(events.borrow_mut().push(event));
 		self.process_pending_events(&event_handler);
-		events.into_inner()
+		let collected_events = events.into_inner();
+
+		// To expand the coverage and make sure all events are properly serialised and deserialised,
+		// we test all generated events round-trip:
+		for event in &collected_events {
+			let ser = event.encode();
+			if let Some(deser) = events::Event::read(&mut &ser[..]).expect("event should deserialize") {
```

Comment:
Is the None case ok to ignore?

Reply:
Yeah, some events are never written so I believe @shaavan is correct here

Reply:
So could also continue when `ser` is None then, and unwrap the read? Not that it is all that important ofc.

Reply:
`ser` just returns `Ok(())` on success, not an option. Also we do currently unwrap the read with the `expect`.. But if you want to rearrange this code somehow, no strong preference here.

Reply:
It looks like `ser` is just `Vec<u8>`. So if that is empty, we expect `None` from `read`? Maybe that can be used to make it slightly stricter and detect the case where read is unexpectedly returning None.

Reply:
Oh right, sorry. It looks like for events where we don't write the actual event, we will still write the type for the event: https://github.com/lightningdevkit/rust-lightning/blob/main/lightning/src/events/mod.rs#L1809 so I don't think we can check if it's empty but we could check if only a u8 was written.

Reply:
Perhaps a too much internal detail then. Can leave this as is as far as I am concerned.